### ISSUE TITLE
Facet max. size cannot be zero (although it can)

### ISSUE THREAD

**Issue by OvalMedia** [Smile-SA/elasticsuite#2078]
Setting the value "Facet max. size" in an attribute's settings the form cannot be submitted if the value in that field is 0.
I can empty the field and submit ok which results in the field acutally being set to 0. A subsequent submission will result in an error message "Please enter a number greater than 0 in this field.".

### Preconditions
PHP 7.3

Magento Version : 2.4.2 (CE)

ElasticSuite Version : latest

Environment : dev

### Steps to reproduce

1. Go to stores -> attributes -> product and pick any attribute
2. Enter attribute's edit form
3. set "Facet max. size" to "0"
4. Hit "Save Attribute"

### Expected result

1. The form should submit ok and save the values to database

### Actual result

1. Submission halts with validation error "Please enter a number greater than 0 in this field."

Basically the validator should handle 0 and an empty field the same. Or it should not automatically set an empty field to 0.

**Comment by vahonc** [comment]
Hello @OvalMedia,

Basically, the Facet max size is the maximum number of elements that will be displayed by a filter in the front office. In other words above it, we will display a "Show more" button. I don't see in which context people would want to set it to 0.

BR,
Vadym

**Comment by OvalMedia** [comment]
0 (or empty) will remove the limit. It will show all items, which is the context I want it to be 0 in.

**Comment by romainruaud** [comment]
Okay we can change the validator to support numbers including 0.

@vahonc can you please propose a PR for this ?

**Comment by vahonc** [comment]
@romainruaud, okay I'll create PR for this one.
